main/src/main/MillServerMain.scala

```diff
@@ -262,4 +268,10 @@ object Server {
       interrupt = false
     }
   }
+
+  def md5hex(str: String): String =
+    hexArray(MessageDigest.getInstance("md5").digest(str.getBytes(StandardCharsets.UTF_8)))
+
+  private def hexArray(arr: Array[Byte]) =
+    String.format("%0" + (arr.length << 1) + "x", new BigInteger(1, arr))
```

Comment:
We should not have two copies of these helper methods. Same methods in class `MillClientMain` are already accessible but currently not visible.